### PR TITLE
[HT-1] Core concepts

### PR DESCRIPTION
**Pull request by maria788**
image

### CODE REVIEW

File: netflix-roulette/src/components/AppHome/components/MoviesList/MoviesList.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+interface MoviesListProps {
+  movies: Movie[];
+}
+
+export class MoviesList extends React.Component<MoviesListProps, {}> {
+  render() {
+    return (
+      <List component="ul">
+        {this.props.movies.map((movie) => (
+          <ListItem>{movie.title}</ListItem>
```
Comment (mylostandromeda) [inline netflix-roulette/src/components/AppHome/components/MoviesList/MoviesList.tsx]: you should provide a key attribute for list items

File: netflix-roulette/src/components/AppHome/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+interface CounterState {
+  searchValue: string;
+}
+
+export class SearchBar extends React.Component<CounterProps, CounterState> {
+  state = { searchValue: "" };
+
+  handleInputChange(event: any) {
```
Comment (mylostandromeda) [inline netflix-roulette/src/components/AppHome/components/SearchBar/SearchBar.tsx]: event type should be `React.ChangeEvent<HTMLInputElement>` but not `any`

File: netflix-roulette/src/components/AppHome/components/MoviesList/MoviesList.tsx (hunk 1 above)
Comment (maria788) [reply netflix-roulette/src/components/AppHome/components/MoviesList/MoviesList.tsx]: Good point, updated

File: netflix-roulette/src/components/AppHome/components/SearchBar/SearchBar.tsx (hunk 2 above)
Comment (maria788) [reply netflix-roulette/src/components/AppHome/components/SearchBar/SearchBar.tsx]: Updated, thanks

File: netflix-roulette/src/components/AppHome/AppHome.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  filteredMovies: Movie[];
+  selectedGenre: string;
+}
+
+export class AppHome extends React.Component<{}, AppHomeState> {
+  state = { filteredMovies: MOVIES, selectedGenre: GenreType.ALL };
+
+  handleSearch(searchText: string) {
+    const filteredMovies = MOVIES.filter((movie) =>
+      movie.title.includes(searchText)
```
Comment (ygorbunkov) [inline netflix-roulette/src/components/AppHome/AppHome.tsx]: I would say, case-insensitive search would deliver slightly better UX:
```ts
handleSearch(searchText: string) {
    const filteredMovies = MOVIES.filter((movie) =>
      movie.title?.toLowerCase().includes(searchText.toLowerCase())
     );
    this.setState({ filteredMovies });
  }
```

File: netflix-roulette/src/components/AppHome/components/GenresTabs/GenresTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import * as React from "react";
+import { Tabs, Tab } from "@material-ui/core";
+import { GENRES } from "../../../../utils";
+
+interface GenresTabsProps {
+  selectedGenre: string;
+  changeGenre(event: React.ChangeEvent<{}>, genre: string): void;
+}
+
+export const GenresTabs: React.FC<GenresTabsProps> = ({
```
Comment (ygorbunkov) [inline netflix-roulette/src/components/AppHome/components/GenresTabs/GenresTabs.tsx]: Using `React.FC` [may be considered a bad practice](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components).

File: netflix-roulette/src/utils/constants.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
+export enum GenreType {
+    DOCUMENTARY = "Documentary",
+    COMEDY = "Comedy",
+    HORROR = "Horror",
+    CRIME = "Crime",
+    ALL = "All",
+}
+
+export const GENRES = [GenreType.ALL, GenreType.DOCUMENTARY, GenreType.COMEDY, GenreType.HORROR, GenreType.CRIME];
```
Comment (ygorbunkov) [inline netflix-roulette/src/utils/constants.ts]: Just for your notice, you may achieve the same with:
```ts
export const GENRES = Object.values(GenreType)
```

File: netflix-roulette/src/components/AppHome/AppHome.tsx (hunk 3 above)
Comment (maria788) [reply netflix-roulette/src/components/AppHome/AppHome.tsx]: Good point. Also added trim to remove extra spaces

File: netflix-roulette/src/utils/constants.ts (hunk 5 above)
Comment (maria788) [reply netflix-roulette/src/utils/constants.ts]: Thanks, updated

**Pull request merged**